packages/perennial-vaults/contracts/balanced/BalancedVaultDefinition.sol

```diff
@@ -87,10 +89,17 @@ contract BalancedVaultDefinition is IBalancedVaultDefinition {
         weight1 = (totalMarkets_ > 1) ? marketDefinitions_[1].weight : DEFAULT_WEIGHT;
 
         for (uint256 marketId; marketId < totalMarkets_; marketId++) {
+            if (marketDefinitions_[marketId].long == marketDefinitions_[marketId].short) revert BalancedVaultDefinitionLongAndShortAreSameProductError();
+            if (!controller.isProduct(marketDefinitions_[marketId].long)) revert BalancedVaultInvalidProductError(marketDefinitions_[marketId].long);
+            if (!controller.isProduct(marketDefinitions_[marketId].short)) revert BalancedVaultInvalidProductError(marketDefinitions_[marketId].short);
+            if (marketDefinitions_[marketId].long.oracle() != marketDefinitions_[marketId].short.oracle()) revert BalancedVaultDefinitionOracleMismatchError();
```

Comment:
you can add one more check here to validate that the payoff is also the inverse for these two:
- the long market has a "long" payoff, and the short market has a "short payoff"
- the "data" for both payoffs is equal (should cover any type)